pkg/controller/servingcert/controller/secret_updating_controller.go

```diff
@@ -114,6 +123,20 @@ func (sc *serviceServingCertUpdateController) requiresRegeneration(secret *v1.Se
 		return true, sharedService
 	}
 
+	// regenerate if secret has bad data key
+	// this checks that the right keys are there
+	// the updateSecret checks the values against known good values
+	for k, _ := range secret.Data {
+		if k != v1.TLSCertKey {
+			if k != v1.TLSPrivateKeyKey {
+				return true, sharedService
+			}
+		}
+	}
+	if len(secret.Data) != 2 {
```

Comment:
With the change to updateSecret() I don't think we can hit these conditions. updateSecret() makes it so that on update if there is an additional data item or someone fudges with the crt/key, the existing data is kept, so we make them immutable on those conditions.  What you added here wants to regenerate on those conditions. Thinking about regeneration vs. protection of the data, it would be better to have protection for the secret update controller and leave the intentional regeneration to the secret creating controller when you take the action of deleting the secret.

Reply:
If I edit an existing secret with bad data(anything different), updateSecret() does not result in that change being set back to the old(good) data.  (I'm probably missing something, but the change in updateSecret() is not enough - regenerating is the only way I've found to get back the secret w/ good data).

Reply:
refactored this, checking/replacing/regenerating as needed - my this PR is getting messy